platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileWriter.java

```diff
@@ -88,24 +84,21 @@ public final class DataFileWriter<D> {
      * @param filePrefix string prefix for all files, must not contain "_" chars
      * @param dataFileDir the path to directory to create the data file in
      * @param index the index number for this file
-     * @param dataItemSerializer Serializer for converting raw data to/from data items
      * @param creationTime the time stamp for the creation time for this file
      */
     public DataFileWriter(
             final String filePrefix,
             final Path dataFileDir,
             final int index,
-            final BaseSerializer<D> dataItemSerializer,
             final Instant creationTime,
             final int compactionLevel)
             throws IOException {
-        this.dataItemSerializer = dataItemSerializer;
         this.path = createDataFilePath(filePrefix, dataFileDir, index, creationTime, DataFileCommon.FILE_EXTENSION);
         metadata = new DataFileMetadata(
                 0, // data item count will be updated later in finishWriting()
                 index,
                 creationTime,
-                dataItemSerializer.getCurrentDataVersion(),
+                0, // backwards compatibility: data version
```

Comment:
Minor: do we have a plan to remove all the "backwards compatibility" reads/writes that we're still performing in this fix in various places?

Reply:
I think this field, `DataFileMetadata.serializationVersion`, can be removed altogether

Reply:
My question is also about a couple of similar places in this fix.

Reply:
We can do the cleanup in a separate PR/project if you like. I'd just want to ensure we do have a plan for the cleanup.